runtime/server/server.go

```diff
@@ -246,3 +259,74 @@ func mapGRPCError(err error) error {
 	}
 	return err
 }
+
+type serverStream struct {
+	grpc.ServerStream
+	ctx context.Context
+}
+
+func (w *serverStream) Context() context.Context {
+	return w.ctx
+}
+
+func wrapServerStream(ctx context.Context, ss grpc.ServerStream) *serverStream {
+	return &serverStream{
+		ServerStream: ss,
+		ctx:          ctx,
+	}
+}
```

Comment:
In other interceptors, we use the `grpc_middleware` package for this. You can replace it with just:
```
wrapped := grpc_middleware.WrapServerStream(ss)
wrapped.WrappedContext = newCtx

return handler(srv, wrapped)
```